lib/ukmpi/isr.c

```diff
@@ -0,0 +1,74 @@
+#include <uk/mbox.h>
+#include <uk/isr/mbox.h>
+
+struct uk_mbox {
```

Comment:
In order to avoid duplicates, it would be much better to place the struct definition in a shared and library-internal header that is used by `mbox.c` and `mbox_isr.c`. You can place it next to the source files, e.g.,
```c
#include "mbox_defs.h"
```

Reply:
Any duplicate function (like `_do_mbox_recv()` could be placed as `static inline` function within this header, too.